java/arcs/core/data/Plan.kt

```diff
@@ -49,6 +50,13 @@ open class Plan(
         }
     }
 
+    /** A [Handle] representation for the [Plan]. */
+    data class Handle(
+        val storageKey: StorageKey,
+        val type: Type,
+        val annotations: List<Annotation>
```

Comment:
If We only plan to allow adapters on connections, than this is ok, otherwise if we do plan to allow adapters on handles themselves, you might want to toss in a nullable Adapter field

Reply:
Things will be easier if we start with adapters just on connections I feel. If we allow them on handles straight away, then we will hit the problem of composing adapters, which is not quite trivial (tractable, but not trivial) while we don't really need it yet. WDYT?

Reply:
SGTM